### PR TITLE
Refactor Mac docs for clarity

### PR DESCRIPTION
**Pull request by bradcray**
As a result in some confusion expressed in issue #16269, I've taken
a pass over the Mac documentation to try to clarify it.  I've also made
changes to the install-mac.html webpage in a separate effort to
avoid sending people to the Quickstart instructions prematurely,
which suggest they'll need to build from source.  Here's what the net
result looks like:

web page:

Screen Shot 2020-09-02 at 6 13 40 PM

docs page:

Screen Shot 2020-09-02 at 6 12 38 PM
Screen Shot 2020-09-02 at 6 12 47 PM

I spent some time thinking about what it would take to eliminate the arguable redundancy between these two pages, but ended up keeping it because I think the right way to get rid of the redundancy would be to eliminate the install-mac.html web page and have it point directly at the Mac-specific docs; but I think that this has the downside of not having a short-and-sweet version of the installation instructions for people who don't need much hand-holding (say, experienced Chapel + homebrew users trying it on a Mac for the first time).

### CODE REVIEW

File: doc/rst/platforms/macosx.rst [hunk 1]
```diff
@@ -4,40 +4,57 @@
 Using Chapel on Mac OS X
 ========================
 
-By and large, Chapel can be used on a Macintosh running OS X just like
-any other platform -- refer to the :ref:`chapelhome-quickstart` for more
-information on building and using Chapel.
+There are two main approaches for using Chapel on Mac OS X:
+
+1) Install via Homebrew_.  For existing Homebrew users, this is the
```
Comment (e-kayrakli) [inline doc/rst/platforms/macosx.rst:9]: drop "existing"?

File: doc/rst/platforms/macosx.rst [hunk 2]
```diff
@@ -4,40 +4,57 @@
...
 
-    /usr/local/Cellar/chapel/<chapel-version>/libexec/
+    chpl /usr/local/Cellar/chapel/<chapel-version>/libexec/hello.chpl
+    ./hello
+
+If you're new to Chapel, refer to the `What's Next?
+<https://chapel-lang.org/docs/usingchapel/QUICKSTART.html#what-s-next>`_
+section of :ref:`chapelhome-quickstart` for next steps.
+    
+.. note::
```
Comment (e-kayrakli) [inline doc/rst/platforms/macosx.rst:42]: This is a bit repetitive, but if you think it is important to emphasize, it is fine

**Review by e-kayrakli — APPROVED** [review]

File: doc/rst/platforms/macosx.rst (hunk 1 above)
Comment (bradcray) [reply doc/rst/platforms/macosx.rst:9]: "current Homebrew users"?  My thought was that if you don't already use homebrew, getting started with homebrew and then using it to install Chapel may be a similar amount of work to installing from source with an inferior result...

File: doc/rst/platforms/macosx.rst (hunk 2 above)
Comment (bradcray) [reply doc/rst/platforms/macosx.rst:42]: Repetitive in that it links to the Quickstart doc and the what's next section of the Quickstart doc you mean?

File: doc/rst/platforms/macosx.rst (hunk 1 above)
Comment (e-kayrakli) [reply doc/rst/platforms/macosx.rst:9]: Yeah maybe. I was just saying that existing/current etc doesn't really add much there. But no big deal.

File: doc/rst/platforms/macosx.rst (hunk 2 above)
Comment (e-kayrakli) [reply doc/rst/platforms/macosx.rst:42]: I actually meant it repeats what we said in the beginning of this doc. i.e. that you don't get multilocale Chapel if you install via `brew`.

File: doc/rst/platforms/macosx.rst (hunk 1 above)
Comment (bradcray) [reply doc/rst/platforms/macosx.rst:9]: Oh, now I see.  Makes sense, thanks.

File: doc/rst/platforms/macosx.rst (hunk 2 above)
Comment (bradcray) [reply doc/rst/platforms/macosx.rst:42]: Oh sorry for my confusion... GitHub's view of the excerpt you commented on emphasized the previous sentences, not the note you were referring to.  I understand now.  I think having it in a bold note covers our butts a bit more (I feel like I still hear people express surprise that their homebrew doesn't support multi-locale... not that this raises the chance they'll read it by much, but...).  Thanks!

Comment (e-kayrakli) [reply doc/rst/platforms/macosx.rst:42]: Yeah that makes sense.

File: doc/rst/platforms/macosx.rst [hunk 3]
```diff
@@ -4,40 +4,57 @@
...
-.. _release: https://github.com/chapel-lang/chapel/releases
+   It is also possible for developers to use Homebrew to build and
+   install a development version based on the master branch on
+   GitHub::
+
+    brew install chapel --HEAD
+
+
+.. _Homebrew: https://brew.sh/
+.. _whatsnext:
```
Comment (cassella) [inline doc/rst/platforms/macosx.rst:60]: Is this "whatsnext" supposed to be here?  I don't see any references to it.

Comment (bradcray) [reply doc/rst/platforms/macosx.rst:60]: Good point, it was a leftover (failed?) attempt that I ended up making a hyperlink within the body of the text itsef.  https://github.com/chapel-lang/chapel/pull/16475 removes it.

**Pull request merged**